src/sds_data_model/_raster.py

```diff
@@ -0,0 +1,466 @@
+        bool: An indication in `_resample_and_reshape whether` to resample data.
+    """
+    if transform.a != expected_cell_size or -transform.e != expected_cell_size:
+        return True
+    else:
+        return False
+
+
+def _has_wrong_shape(
+    transform: Affine,
+    expected_x_min: int,
+    expected_y_max: int,
+) -> bool:
+    """Return true if the shape/extent of the input data \
+    is **not equal** to the expected shape.
+
+    Args:
+        transform (Affine): Input transform, the data being checked.
+        expected_x_min (int): The expected value for the x minimum.
+        expected_y_max (int): The expected value for the y maximum.
+
+    Returns:
+        bool: An indication in `_resample_and_reshape` whether to reshape/regrid data.
+    """
+    if transform.f != expected_y_max or transform.c != expected_x_min:
+        return True
+    else:
+        return False
+
+
+def _create_data_array(
+    data_array: DataArray,
+    data: NDArray,
+    geotransform: str,
+) -> DataArray:
+    """An intermediate `DataArray` constructor for \
+    the updating of shape/extent/transform.
+
+    Args:
+        data_array (DataArray): Original `DataArray`.
+        data (NDArray): New array data.
+        geotransform (str): New (gdal) geotransform.
+
+    Returns:
+        DataArray: `DataArray` with the input data and geotransform added.
+    """
+    return DataArray(
+        data=data,
+        coords={
+            data_array.rio.grid_mapping: data_array[
+                data_array.rio.grid_mapping
+            ].rio.update_attrs({"GeoTransform": geotransform}),
+        },
+        dims=("northings", "eastings"),
+        attrs={"nodata": data_array.rio.nodata},
+    ).rename(data_array.name)
+
+
+def _get_resample_shape(
+    data_array: DataArray,
+    expected_cell_size: int,
+) -> NDArray:
+    """Determine the new (scaled) shape of an input DataArray from a cell_size value.
+
+    Args:
+        data_array (DataArray): Input `DataArray`.
+        expected_cell_size (int): New cell size for data. Acts as the resampling factor.
+
+    Returns:
+        NDArray: 2d array of (height, width).
+    """
+    return (
+        (
+            (
+                array([data_array.rio.transform().a, -data_array.rio.transform().e])
+                / expected_cell_size
+            )
+            * data_array.shape
+        )
+        .round()
+        .astype("int")
+    )
+
+
+def _resample_cell_size(
+    data_array: DataArray,
+    expected_cell_size: int,
+    categorical: bool,
+) -> DataArray:
+    """Returns a `DataArray` with scaled/resampled data using OpenCV.
+
+    Args:
+        data_array (DataArray): Input `DataArray`.
+        expected_cell_size (int): New cell size for data. Acts as the resampling factor.
+        categorical (bool): Determines which resampling/interpolation is used.
+            Set to False will use bilinear. Set to True will use nearest-neighbour.
+
+    Returns:
+        DataArray
+    """
```

Comment:
1. The return type should include a description.
2. You can add a link to the OpenCV project to the docstring like this:

```suggestion
    """Returns a `DataArray` with scaled/resampled data using `OpenCV`_.

    Args:
        data_array (DataArray): Input `DataArray`.
        expected_cell_size (int): New cell size for data. Acts as the resampling factor.
        categorical (bool): Determines which resampling/interpolation is used.
            Set to False will use bilinear. Set to True will use nearest-neighbour.

    Returns:
        DataArray: a `DataArray` with scaled/resampled data.
    
    .. _`OpenCV`:
        https://github.com/opencv/opencv-python    
    """
```